### PR TITLE
feat: snapshot compress: default output, progress bar, overwrite prompt

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- set default output destination to the current directory,
- prompt if we're about to overwrite the destination file (inhibit with `--force`),
- show a progress bar.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/cli/subcommands/snapshot_cmd.rs [hunk 1]
```diff
@@ -194,11 +204,49 @@ impl SnapshotCommands {
...
+                            destination.set_extension("");
+                        }
+                        destination.with_extension("forest.car.zst")
+                    }
+                    false => output.clone(),
+                };
+
+                if destination.exists() && !force {
+                    let have_permission = Confirm::with_theme(&ColorfulTheme::default())
+                        .with_prompt(format!("{:?} will be overwritten. Continue?", destination))
```
Comment (LesnyRumcajs) [inline src/cli/subcommands/snapshot_cmd.rs:231]: Why `Debug` format and not `Display`?

**Review by LesnyRumcajs — APPROVED** [review]

File: src/cli/subcommands/snapshot_cmd.rs (hunk 1 above)
Comment (lemmih) [reply src/cli/subcommands/snapshot_cmd.rs]: Switched to `.to_string_lossy()`.

File: src/cli/subcommands/snapshot_cmd.rs [hunk 2]
```diff
@@ -196,11 +206,52 @@ impl SnapshotCommands {
...
+                        .unwrap_or(false);
+                    if !have_permission {
+                        return Ok(());
+                    }
+                }
+
+                println!("Generating ForestCAR.zst file: {:?}", &destination);
+
+                let file = File::open(&source).await?;
+                let pb = ProgressBar::new(file.metadata().await?.len()).with_style(
```
Comment (hanabi1224) [inline src/cli/subcommands/snapshot_cmd.rs:249]: Just to confirm, should we only use log based progress bar in daemon, and `indicatif` in cli and tools?

**Review by hanabi1224 — APPROVED** [review]

File: src/cli/subcommands/snapshot_cmd.rs (hunk 2 above)
Comment (lemmih) [reply src/cli/subcommands/snapshot_cmd.rs:249]: Yes.

**Pull request merged**